### PR TITLE
gh-91210: Improve error message when non-default param follows default

### PR DESCRIPTION
**Pull request by lysnikolaou**
- Improve error message when parameter without a default follows one with a default
- Show same error message when positional-only params precede the default/non-default sequence

* Issue: gh-91210

Automerge-Triggered-By: GH:lysnikolaou

### CODE REVIEW

**Comment by lysnikolaou** [comment]
@pablogsal Very slight nudge on this.

File: Misc/NEWS.d/next/Core and Builtins/2022-08-12-18-13-49.gh-issue-91210.AWMSLj.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Improve error message when a parameter without a default value follows one with a default value and show the same message, even when the non-default/default sequence is preceded by positional-only parameters.
```
Comment (pablogsal) [inline Misc/NEWS.d/next/Core and Builtins/2022-08-12-18-13-49.gh-issue-91210.AWMSLj.rst]:
```suggestion
Improve error message when a parameter without a default value follows one with a default value and shows the same message, even when the non-default/default sequence is preceded by positional-only parameters.
```

**Review by pablogsal — APPROVED** [review]
LGTM with a small nit on the commit message.

Thanks for working on this 🚀

File: Misc/NEWS.d/next/Core and Builtins/2022-08-12-18-13-49.gh-issue-91210.AWMSLj.rst (hunk 1 above)
Comment (lysnikolaou) [reply Misc/NEWS.d/next/Core and Builtins/2022-08-12-18-13-49.gh-issue-91210.AWMSLj.rst]: Thanks for the review @pablogsal. `show` here refers to the PR fixing a difference in an error, it doesn't describe current behavior. Something like `Improve error message, and show the same message`.

**Comment by lysnikolaou** [comment]
@pablogsal I guess we're not backporting this, correct?

**Comment by pablogsal** [comment]
> @pablogsal I guess we're not backporting this, correct?

Unfortunately no, since it's technically an improvement

**Pull request merged**